charts/skypilot/templates/api-deployment.yaml

```diff
@@ -0,0 +1,191 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: skypilot-api-server
+  namespace: {{ .Release.namespace }}
+spec:
+  replicas: {{ .Values.apiService.replicas }}
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app: skypilot-api
+  template:
+    metadata:
+      labels:
+        app: skypilot-api
+    spec:
+      automountServiceAccountToken: {{ .Values.kubernetesCredentials.useApiServerCluster }}
+      serviceAccountName: skypilot-api-sa
+      containers:
+      - name: skypilot-api
+        image: {{ .Values.apiService.image }}
+        imagePullPolicy: Always
+        resources:
+          {{- if .Values.apiService.cpuCoresLimit }}
+          limits:
+            cpu: {{ .Values.apiService.cpuCoresLimit }}
+          {{- end }}
+          {{- if .Values.apiService.memoryGBLimit }}
+            memory: {{ .Values.apiService.memoryGBLimit }}G
+          {{- end }}
+        env:
+        - name: SKYPILOT_DEV
+          value: {{ .Values.apiService.skypilotDev | quote }}          
+        {{- if .Values.apiService.cpuCoresLimit }}
+        - name: SKYPILOT_POD_CPU_CORES_LIMIT
+          value: {{ .Values.apiService.cpuCoresLimit | quote }}
+        {{- end }}
+        {{- if .Values.apiService.memoryGBLimit }}
+        - name: SKYPILOT_POD_MEMORY_GB_LIMIT
+          value: {{ .Values.apiService.memoryGBLimit | quote }}
+        {{- end }}
+        {{- if .Values.gcpCredentials.enabled }}
+        - name: GOOGLE_APPLICATION_CREDENTIALS
+          value: /root/gcp-cred.json
+        {{- end }}
+        {{- if .Values.kubernetesCredentials.inclusterNamespace }}
+        - name: SKYPILOT_IN_CLUSTER_NAMESPACE
+          value: {{ .Values.kubernetesCredentials.inclusterNamespace }}
+        {{- end }}
+        # Start api service and tail logs
+        command: ["/bin/sh", "-c"]
```

Comment:
IIRC, `sh` does not handle signals sent to pid 1 properly, using https://github.com/Yelp/dumb-init as entrypoint would help handling signals like SIGTERM and reaping zombie process

Reply:
great catch, added now